W-14145549-parse-template-cv (#180)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/mule-components/4.4/component-parse-template.adoc

//
//tag::component-title[]

= Parse Template (<parse-template/>)

//end::component-title[]
//

//
//tag::component-short-description[]
//     Short description of the form "Do something..." 
//     Example: "Configure log messages anywhere in a flow."

Process a template and obtain a result.

//end::component-short-description[]
//

//
//tag::component-long-description[]

> A template is defined as text with embedded DataWeave expressions that Mule evaluates and replaces with their result.

Reviewer: ```suggestion
A template includes text with embedded DataWeave expressions that Mule evaluates and replaces with their result.
```

> Configure a template through an external file reference or you can embed the template in the component definition.

Reviewer: ```suggestion
Configure a template through an external file reference, or embed the template in the component configuration.
```

> You can configure a template through an external file reference or you can embed the template in the component definition.

Reviewer: ```suggestion
Configure a template through an external file reference or you can embed the template in the component definition.
```

//end::component-long-description[]
//

//SECTION: COMPONENT XML
//
//tag::component-xml-title[]

[[component-xml]]
== Component XML

This component supports the following XML structure:

//end::component-xml-title[]
//
//
//tag::component-xml[]

[source,xml]
----
<parse-template 
  doc:name="Parse template" 
  doc:id="fppwuw" 
  location="template.html" />
----

//end::component-xml[]
//
//tag::component-snippet-xml[]

[[snippet]]

=== Snippet XML

Snippet XML in Anypoint Code Builder serves as a customizable template for the component. 

[source,xml]
----
SNIPPET_XML_HERE
----

//end::component-snippet-xml[]
//
//
//
//
//TABLE: ROOT XML ATTRIBUTES (for the top-level (root) element)
//tag::component-xml-attributes-root[]

[%header,cols="1,3a"]
|===
| `<parse-template/>` Attributes 
| Description

| `doc:name` 
| Editable name for the component.

| `doc:id` 
| Automatically generated identifier for the component.

| `location` 

> | Location of the file that Mule uses as a template into which to insert values extracted from the message properties or variables.

Reviewer: ```suggestion
| Location of the template file. Mule inserts values extracted from the message properties or variables into this file.
```

| `content`

> | String representing the template to be used where the embedded expressions will be evaluated and replaced by their results.

Reviewer: ```suggestion
| String representing the template where embedded expressions are evaluated and replaced by their results.
```

| `target`

> | Name of the variable where the result of the expression defined in `targetValue` is stored after Parse Template executes.

Reviewer: ```suggestion
| Name of the variable where the result of the expression defined in `targetValue` is stored after Parse Template executes. For more information, see xref:mule-runtime::target-variables.adoc[].
```

> | Name of the variable where the result of the expression defined in `targetValue` is stored after the Parse Template is executed.

Reviewer: ```suggestion
| Name of the variable where the result of the expression defined in `targetValue` is stored after Parse Template executes.
```
